transport/ws/src/test/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactoryTest.java

```diff
@@ -0,0 +1,272 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaazing.gateway.transport.ws.extension;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.net.ProtocolException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.jmock.Expectations;
+import org.jmock.Mockery;
+import org.jmock.lib.legacy.ClassImposteriser;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.kaazing.gateway.resource.address.ws.WsResourceAddress;
+import org.kaazing.gateway.transport.http.HttpAcceptSession;
+import org.kaazing.gateway.transport.ws.extension.WebSocketExtensionFactoryTest.MockWebSocketExtensionSpi.MockNegotiate;
+
+public class WebSocketExtensionFactoryTest {
+
+    private WebSocketExtensionFactory wsExtFactory;
+    private WsResourceAddress address;
+    private ExtensionHeader extensionHeader;
+    private MockNegotiate mockNegotiate;
+    private Mockery context;
+    private WebSocketExtensionSpi webSocketExtensionSpi;
+    private HttpAcceptSession session;
+
+    @Before
+    public void init() {
+        wsExtFactory = WebSocketExtensionFactory.newInstance();
+        context = new Mockery() {
+            {
+                setImposteriser(ClassImposteriser.INSTANCE);
+            }
+        };
+        address = context.mock(WsResourceAddress.class);
+        extensionHeader = context.mock(ExtensionHeader.class);
+        mockNegotiate = context.mock(MockNegotiate.class);
+        MockWebSocketExtensionSpi.setNegotiateBehavoir(mockNegotiate);
+        webSocketExtensionSpi = context.mock(WebSocketExtensionSpi.class);
+        session = context.mock(HttpAcceptSession.class);
+    }
+
+    @Test
+    public void testGetExtensionNames() {
+        Collection<String> extensionNames = wsExtFactory.getExtensionNames();
+        assertTrue(extensionNames.contains("mock"));
+    }
+
+    @Test
+    public void testNegotiateExtFound() throws IOException {
+        context.checking(new Expectations() {
+            {
+                oneOf(extensionHeader).getExtensionToken();
+                will(returnValue("mock"));
+                oneOf(mockNegotiate).negotiate(extensionHeader, address);
+                will(returnValue(webSocketExtensionSpi));
+            }
+        });
+
+        WebSocketExtensionSpi extension = wsExtFactory.negotiate(extensionHeader, address);
+        Assert.assertNotNull(extension);
+        Assert.assertSame(extension, webSocketExtensionSpi);
+        context.assertIsSatisfied();
```

Comment:
You can define a Rule as follows to have the Mockery be created and assertIsSatisfied for each test method:
@Rule
public JUnitRuleMockery context = new JUnitRuleMockery() {{
   setImposteriser(ClassImposteriser.INSTANCE);
}};
// Use context as Mockery. It is automatically checked (assertIsSatisfied) after each @Test method.